include/zephyr/usb/usb_device.h

```diff
@@ -79,7 +79,13 @@ extern "C" {
  *  USB configuration
  **************************************************************************/
 
-#define USB_MAX_CTRL_MPS	64   /**< maximum packet size (MPS) for EP 0 */
+/**< maximum packet size (MPS) for EP 0 */
+#if DT_NODE_HAS_PROP(DT_NODELABEL(usbd), ep_out_buf_size)
+	#define USB_MAX_CTRL_MPS	DT_PROP_BY_IDX(DT_NODELABEL(usbd), ep_out_buf_size, 0)
+#else
+	#define USB_MAX_CTRL_MPS	64   
+#endif
+
```

Comment:
This is USB device stack header. You pull a DT macro to a subsystem to get a property of the controller. I will not accept dependencies on any DT macros or Kconfig options in this header. I was probably not clear enough in [that comment](https://github.com/zephyrproject-rtos/zephyr/pull/52278#issuecomment-1401895950), I apologize. To be clear (also fro #56838), the correct way would be to let the stack get control MPS information using the usb_dc driver API, see https://github.com/zephyrproject-rtos/zephyr/pull/57028